lib/src/http/common/request.dart

```diff
@@ -707,4 +704,49 @@ abstract class CommonRequest extends Object
     didSucceed = true;
     return response;
   }
+
+  Duration _createExponentialBackOff() {
+    int backOffInMs = autoRetry.backOff.interval.inMilliseconds *
+        pow(defaultExponentialMultiplier, autoRetry.numAttempts);
+    backOffInMs =
+        min(autoRetry.backOff.maxInterval.inMilliseconds, backOffInMs);
+
+    if (autoRetry.backOff.withJitter == true) {
+      Random random = new Random();
+      backOffInMs = random.nextInt(backOffInMs);
+    }
+    return new Duration(milliseconds: backOffInMs);
+  }
+
+  Duration _createFixedBackOff() {
```

Comment:
Could you move this method and the one above out into a utility file? There's no hard dependency on this class, they could be updated to accept `interval`, `maxInterval`, and `withJitter` as parameters. Then you can unit test this math without needing to deal with requests.